### PR TITLE
Enhance security2 with PersmissionService

### PR DESCRIPTION
**Pull request by christinaspachiou**

### CODE REVIEW

File: security2/src/main/java/zone/cogni/asquare/security2/SecurityController.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import zone.cogni.asquare.security2.model.AuthenticationToken;
+import zone.cogni.asquare.security2.model.AuthenticationTokenMapper;
+import zone.cogni.asquare.security2.model.UserDto;
+
+@RestController
+@RequestMapping("/api/public")
+public class SecurityController {
```
Comment (domivds) [inline security2/src/main/java/zone/cogni/asquare/security2/SecurityController.java:14]: Lib should not contain Controller class (that should happen only in very rare cases)

File: security2/src/main/java/zone/cogni/asquare/security2/model/AuthenticationToken.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.Set;
+
+@JsonSerialize(
+  using = AuthenticationToken.TokenJsonSerializer.class
+)
+public class AuthenticationToken extends AbstractAuthenticationToken {
```
Comment (domivds) [inline security2/src/main/java/zone/cogni/asquare/security2/model/AuthenticationToken.java:21]: Not needed in PermissionService only Controller, so can be removed

File: security2/src/main/java/zone/cogni/asquare/security2/model/AuthenticationTokenMapper.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package zone.cogni.asquare.security2.model;
+
+public class AuthenticationTokenMapper {
```
Comment (domivds) [inline security2/src/main/java/zone/cogni/asquare/security2/model/AuthenticationTokenMapper.java:3]: Not needed in PermissionService only Controller, so can be removed

File: security2/src/main/java/zone/cogni/asquare/security2/model/UserDto.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
+package zone.cogni.asquare.security2.model;
+
+import java.util.Set;
+
+public class UserDto {
```
Comment (domivds) [inline security2/src/main/java/zone/cogni/asquare/security2/model/UserDto.java:5]: Not needed in PermissionService only Controller, so can be removed

File: security2/src/main/java/zone/cogni/asquare/security2/service/PermissionService.java [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class PermissionService<T extends Enum> {
+  private static final Logger log = LoggerFactory.getLogger(PermissionService.class);
+  private final Map<String, Set<String>> blacklistProperties = new HashMap<>();
+  private final Map<String, Set<String>> notBlacklistedForRole = new HashMap<>();
```
Comment (domivds) [inline security2/src/main/java/zone/cogni/asquare/security2/service/PermissionService.java:26]: I don't think this is used anywhere at the end. Also this should not be in this class. 
So to be removed (with methods using it)

File: security2/src/main/java/zone/cogni/asquare/security2/service/PermissionService.java [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+  private final Map<String, String> ldap2roleName = new HashMap<>();
```
Comment (domivds) [inline security2/src/main/java/zone/cogni/asquare/security2/service/PermissionService.java:27]: implementation specific, not good
also only used in init, so remove

File: security2/src/main/java/zone/cogni/asquare/security2/service/PermissionService.java [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+  private final Map<String, Set<T>> roleName2permissions = Collections.synchronizedMap(new HashMap<>());
+  private final Class<T> enumClass;
+  private Set<T> defaultPermissions;
+  private Set<String> allBlacklistProperties;
+  @Value("${asquare.security.role_group.resource:/role-group.json}")
+  private String ldapRoleGroupResource;
```
Comment (domivds) [inline security2/src/main/java/zone/cogni/asquare/security2/service/PermissionService.java:33]: to remove, actually unused

File: security2/src/main/java/zone/cogni/asquare/security2/service/PermissionService.java [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class PermissionService<T extends Enum> {
```
Comment (domivds) [inline security2/src/main/java/zone/cogni/asquare/security2/service/PermissionService.java:23]: global comment for class: add debug logs on the different actions and checks for permissions

File: security2/src/main/java/zone/cogni/asquare/security2/service/PermissionService.java [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+    log.info("Using ldapRoleGroupResource '{}'", ldapRoleGroupResource);
+
+    try (InputStream inputStream = PermissionService.class.getResourceAsStream(ldapRoleGroupResource)) {
+      Map<String, String> jsonData = new ObjectMapper().enable(JsonParser.Feature.ALLOW_COMMENTS).readValue(inputStream, Map.class);
+      for (Map.Entry<String, String> entry : jsonData.entrySet()) {
+        ldap2roleName.put(StringUtils.lowerCase(entry.getKey()), StringUtils.lowerCase(entry.getValue()));
+      }
+    }
+
+    try (InputStream inputStream = PermissionService.class.getResourceAsStream("/role_access.json")) {
```
Comment (domivds) [inline security2/src/main/java/zone/cogni/asquare/security2/service/PermissionService.java:51]: resource location to be specified in config (do not use a default value + use config prefix used in security2 lib)

also add a bit of info log during init

**Comment by christinaspachiou** [comment]
PermissionService moved to https://github.com/cognizone/security

**Pull request closed without merging**